File: pascal/arrays/iterating_over_arrays_and_updating_elements.pas
program IteratingOverArraysAndUpdatingElements;

var
  some_array: array[0..2] of integer = (1, 3, 5);
  i: integer;

begin
  writeln('Before:');
  for i := 0 to high(some_array) do
    writeln('array[', i, ']: ', some_array[i]);

  for i := 0 to high(some_array) do
    some_array[i] *= 2;

  writeln('After:');
  for i := 0 to high(some_array) do
    writeln('array[', i, ']: ', some_array[i]);

  writeln();
end.

File: c#/arrays/IteratingOverArraysAndUpdatingElements.cs
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class IteratingOverArraysAndUpdatingElements {
  public static void Main(string[] args) {
    int[] array = {1, 3, 5};

    System.Console.WriteLine("Before:");
    for (int i = 0; i < array.Length; i++)
      System.Console.WriteLine(string.Format("array[{0}]: {1}", i, array[i]));

    for (int i = 0; i < array.Length; i++)
      array[i] *= 2;

    System.Console.WriteLine("After:");
    for (int i = 0; i < array.Length; i++)
      System.Console.WriteLine(string.Format("array[{0}]: {1}", i, array[i]));

    System.Console.WriteLine();
  }
}
